packages/protocol/contracts/stability/CPExchange.sol

```diff
@@ -0,0 +1,132 @@
+pragma solidity ^0.5.13;
+pragma experimental ABIEncoderV2;
+
+import { IMentoExchange } from "./interfaces/IMentoExchange.sol";
+import { ISortedOracles } from "./interfaces/ISortedOracles.sol";
+import { IReserve } from "./interfaces/IReserve.sol";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { Initializable } from "../common/Initializable.sol";
+import { FixidityLib } from "../common/FixidityLib.sol";
+import { UsingRegistry } from "../common/UsingRegistry.sol";
+import { ReentrancyGuard } from "../common/libraries/ReentrancyGuard.sol";
+
+contract CPExchange is IMentoExchange, UsingRegistry, ReentrancyGuard, Initializable {
+  using SafeMath for uint256;
+
+  /* ==================== Constructor ==================== */
+
+  /**
+   * @notice Sets initialized == true on implementation contracts
+   * @param test Set to true to skip implementation initialization
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice Allows the contract to be upgradable via the proxy.
+   * @param registryAddress The address of the Celo registry.
+   */
+  function initilize(address registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+  }
+
+  /* ==================== View Functions ==================== */
+
+  function getAmountOut(
```

Comment:
Without a `getAmountIn` how can broker retrieve a price for a buy order, i.e. user wants to buy `X` Celo or cSTable?

Reply:
Agreed! Done.